plugins/sentinelone/komand_sentinelone/tasks/monitor_logs/task.py

```diff
@@ -0,0 +1,338 @@
+        is_paginating: bool,
+    ) -> Tuple[List[Dict], int]:
+        """
+        Handle whether to query each endpoint based on input and pagination, gather returned data and error responses
+        """
+        all_logs = []
+        for log_type, check in queries.items():
+            if check:
+                cursor = cursors.get(log_type)
+                if is_paginating is False or (is_paginating is True and cursor):
+                    self.logger.info(f"Getting {log_type} between {lookback_timestamp} and {current_run_timestamp}...")
+                    if cursor:
+                        self.logger.info(f"Using next page cursor: {cursor}")
+                    logs, total_forbidden_responses = self.get_generic_logs(
+                        log_type,
+                        state,
+                        total_forbidden_responses,
+                        current_run_timestamp,
+                        lookback_timestamp,
+                        cursor,
+                    )
+                    self.logger.info(f"{len(logs)} logs found in {log_type} query")
+                    all_logs.extend(logs)
+                else:
+                    self.logger.info(f"Pagination in progress, skipping {log_type} collection...")
+        return all_logs, total_forbidden_responses
+
+    def get_generic_logs(
+        self,
+        log_type: str,
+        state: dict,
+        total_forbidden_responses: int,
+        current_run_timestamp: str,
+        last_run_timestamp: str = None,
+        pagination_cursor: str = None,
+    ) -> List[Dict]:
+        """
+        Determine log type and query that log set with query parameters determined by type, cursor, and last timestamp.
+        Set latest log timestamp and page cursor in state.
+        Return logs from query
+        """
+        timestamp_key = LOG_TYPE_MAP.get(log_type, {}).get("LAST_LOG_TIMESTAMP")
+        page_token_key = LOG_TYPE_MAP.get(log_type, {}).get("PAGE_TOKEN")
+        last_log_timestamp = state.get(timestamp_key)
+        query_params = self.get_query_params(log_type, last_run_timestamp, current_run_timestamp, pagination_cursor)
+        if log_type == ACTIVITIES_LOGS:
+            response = self.connection.client.get_activities_list(query_params, True, False)
+        elif log_type == EVENTS_LOGS:
+            response = self.connection.client.get_device_control_events(query_params, True, False)
+        else:
+            response = self.connection.client.get_threats(query_params, API_VERSION, True, False)
+        try:
+            status_code = response.status_code
+            if response.status_code in [401, 403]:
+                self.logger.info(f"Received status code {status_code} for {log_type} query")
+            response_handler(response, allowed_status_codes=[401, 403])
+        except PluginException as error:
+            raise ApiException(cause=error.cause, assistance=error.assistance, status_code=status_code, data=response)
```

Comment:
Log this error?

Reply:
I've bubbled up error data to to log in the run function.